core/src/main/java/org/carbondata/scan/collector/impl/AbstractScannedResultCollector.java

```diff
@@ -170,33 +120,25 @@ private Object getMeasureData(MeasureColumnDataChunk dataChunk, int index, DataT
   /**
    * Below method will used to get the result
    */
-  @Override public Result getCollectedResult() {
-    Result<List<ListBasedResultWrapper>, Object> result = new ListBasedResult();
-    if (tableBlockExecutionInfos.isFixedKeyUpdateRequired() && tableBlockExecutionInfos
-        .isDimensionsExistInQuery()) {
-      updateKeyWithLatestBlockKeygenerator();
-      result.addScannedResult(listBasedResult);
-    } else {
-      result.addScannedResult(listBasedResult);
+  protected void updateData(List<Object[]> listBasedResult) {
+    if (tableBlockExecutionInfos.isFixedKeyUpdateRequired()) {
+      updateKeyWithLatestBlockKeygenerator(listBasedResult);
     }
-    return result;
   }
 
-
-
   /**
    * Below method will be used to update the fixed length key with the
    * latest block key generator
    *
    * @return updated block
    */
-  private void updateKeyWithLatestBlockKeygenerator() {
+  private void updateKeyWithLatestBlockKeygenerator(List<Object[]> listBasedResult) {
```

Comment:
@kumarvishal09 Yes you are right, as we are filling with surrogates we don't require to update/restructure the key. Key update/restructure is not required only for `DictionaryBasedResultCollector` but it is required for `RawBasedResultCollector`. That is why it is added here.